File: Cap 07/Topico 7.3/Pesquisa Sequencial/Numerica/pesqseqnum.scala
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.3
 *
 * Programa ..: PESQUISA SEQUENCIAL POR NUMERO
 */

object Main {
  def main(args: Array[String]): Unit = {
    
    val NUMERO: Array[Int] = new Array[Int](5)
    var I: Int = 0
    var PESQ: Int = 0
    var RESP: String = ""
    var ACHA: Boolean = false

    println("PESQUISA SEQUENCIAL DE NUMEROS\n")

    for (I <- 0 to 4 by 1) {
      print(s"Entre o ${I + 1}o. numero: ")
      NUMERO(I) = scala.io.StdIn.readLine().toInt
    }

    RESP = "SIM"
    while (RESP == "SIM") {
      println()
      print("Entre numero a ser pesquisado: ")
      PESQ = scala.io.StdIn.readLine().toInt
      I = 0
      ACHA = false
      while (I <= 4 && ACHA == false) {
        if (PESQ == NUMERO(I)) {
          ACHA = true
        } else {
          I = I + 1
        }
      }
      if (ACHA == true) {
        println()
        println(s"$PESQ foi localizado na posicao ${I + 1}")
      } else {
        println()
        println(s"$PESQ nao foi localizado")
      }
      println()
      print("Deseja continuar? (SIM/NAO): ")
      RESP = scala.io.StdIn.readLine().toUpperCase
    }
    
  }
}

File: Cap 07/Topico 7.3/Pesquisa Sequencial/Numerica/pesqseqnum.php
<?php
/*
 * Livro .....: Algoritmos
 *              Lógica para Desenvolvimento de Programação de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 7 - Topico 7.3
 * 
 * Programa ..: PESQUISA SEQUENCIAL POR NUMERO
 */

echo "PESQUISA SEQUENCIAL DE NUMEROS\n\n";

$NUMERO = array();

for ($I = 1; $I <= 5; $I++) {
  echo "Entre o " . $I . "o. numero: ";
  $NUMERO[$I] = readline();
}

// *** inicio do trecho de pesquisa sequencial ***

$RESP = 'SIM';

while ($RESP === 'SIM') {
  echo "\n";
  echo "Entre o numero a ser pesquisado: ";
  $PESQ = readline();
  $I = 1;
  $ACHA = false;
  while ($I <= 5 && $ACHA === false) {
    if ($PESQ === $NUMERO[$I]) {
      $ACHA = true;
    } else {
      $I++;
    }
  }
  echo "\n";
  if ($ACHA === true) {
    echo $PESQ . " foi localizado na posicao " . $I . "\n";
  } else {
    echo $PESQ . " nao foi localizado\n";
  }
  echo "\n";
  echo "Deseja continuar? (SIM/NAO): ";
  $RESP = readline();
  $RESP = strtoupper($RESP);
}

// *** fim do trecho de pesquisa sequencial ***

?>
